### PR TITLE
Fix Seq Validation

### PR DESCRIPTION
**Pull request by jackdingilian**
The current approach to Seq validation fails on Seqs containing nested records when it tries to get the ValidationType name from the nested record. It needs the ValidationType name in order to publish metrics. To avoid this problem this splits the field level validation logic into its own function that writes the metrics. This can then be re-used from `DerivedValidator` and `SeqLikeValidator` against any `FieldValidator`

### CODE REVIEW

File: elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala [hunk 1]
```diff
@@ -16,9 +16,14 @@
  */
 package com.spotify.elitzur.validators
 
-import java.lang.{StringBuilder => JStringBuilder}
+import com.spotify.elitzur.validators.Validator.validateField
 
-import com.spotify.elitzur.{DataInvalidException, IllegalValidationException, MetricsReporter}
+import java.lang.{StringBuilder => JStringBuilder}
+import com.spotify.elitzur.{
```
Comment (jackdingilian) [inline elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala:22]: Sorry, running scalafmt on this made it seem like a much larger change. The material changes are in `SeqLikeValidator` (line 184),  `validationLoop` (line 327), and `validateField` (line 424)

File: elitzur-core/src/test/scala/com/spotify/elitzur/validators/ValidatorTest.scala [hunk 2]
```diff
@@ -0,0 +1,207 @@
+package com.spotify.elitzur.validators
```
Comment (jackdingilian) [inline elitzur-core/src/test/scala/com/spotify/elitzur/validators/ValidatorTest.scala:1]: I think some of our core tests got dropped when we open sourced. Adding some simple testing here so we aren't relying on ValidatorDoFnTest as much

File: benchmarking/src/main/scala/com/spotify/elitzur/Benchmarking.scala [hunk 3]
```diff
@@ -125,6 +125,7 @@ object CaseClassValidators {
   val fiveNFiveVal: Validator[FiveNestedFiveFields] = genFiveNestedFive()
 }
 
+//scalastyle:off magic.number
```
Comment (idreeskhan) [inline benchmarking/src/main/scala/com/spotify/elitzur/Benchmarking.scala:128]: Need a correspond on toggle

**Review by idreeskhan — APPROVED** [review]
Hard to parse the details with the scalafmt changes but looks intuitive to me
